ui/pages/confirmations/hooks/useConfirmationAlertMetrics.ts

```diff
@@ -58,75 +61,72 @@ export function useConfirmationAlertMetrics() {
-          : ALERTS_NAME_METRICS[alertKey] ?? alertKey;
+  const trackAlertRender = useCallback((alertKey: string) => {
+    setMetricsProperties((prevState) => {
+      const newState = { ...prevState };
+      const alertName = getAlertName(alertKey);
+      newState.alert_visualized = uniqueFreshArrayPush(
+        prevState.alert_visualized,
+        alertName,
+      );
+      newState.alert_visualized_count = newState.alert_visualized.length;
+      return newState;
+    });
+  }, []);
 
-        switch (action) {
-          case AlertsActionMetrics.AlertVisualized:
-            newState.alert_visualized = [
-              ...new Set([...prevState.alert_visualized, alertName]),
-            ];
-            newState.alert_visualized_count = newState.alert_visualized.length;
-            break;
-          case AlertsActionMetrics.InlineAlertClicked:
-            newState.alert_key_clicked = [
-              ...new Set([...prevState.alert_key_clicked, alertName]),
-            ];
-            break;
-          case AlertsActionMetrics.AlertActionClicked:
-            newState.alert_action_clicked = [
-              ...new Set([...prevState.alert_action_clicked, alertName]),
-            ];
-            break;
-          default:
-        }
-        return newState;
-      });
-    },
-    [isValidType],
-  );
+  const trackInlineAlertClicked = useCallback((alertKey: string) => {
+    setMetricsProperties((prevState) => {
+      const newState = { ...prevState };
+      const alertName = getAlertName(alertKey);
+      newState.alert_key_clicked = uniqueFreshArrayPush(
+        prevState.alert_key_clicked,
+        alertName,
+      );
+      return newState;
+    });
+  }, []);
 
-  useEffect(() => {
-    if (isValidType && alerts.length > 0) {
-      const properties = {
-        alert_triggered_count: alerts.length,
-        alert_triggered: getAlertsName(alerts),
-        alert_resolved_count: alerts.filter((alert) =>
-          isAlertConfirmed(alert.key),
-        ).length,
-        alert_resolved: getAlertsName(
-          alerts.filter((alert) => isAlertConfirmed(alert.key)),
-        ),
-        ...metricsProperties,
-      };
-      updateTransactionEventFragment({ properties }, ownerId);
+  const trackAlertActionClicked = useCallback((alertKey: string) => {
+    setMetricsProperties((prevState) => {
+      const newState = { ...prevState };
+      const alertName = getAlertName(alertKey);
+      newState.alert_action_clicked = uniqueFreshArrayPush(
+        prevState.alert_action_clicked,
+        alertName,
+      );
+      return newState;
+    });
+  }, []);
+
+  const updateAlertMetrics = useCallback(() => {
+    if (!properties) {
+      return;
     }
-  }, [
-    metricsProperties,
-    alerts,
-    isValidType,
-    ownerId,
-    updateTransactionEventFragment,
-  ]);
+    updateTransactionEventFragment({ properties }, ownerId);
+  }, [properties, updateTransactionEventFragment, ownerId]);
```

Comment:
We're building a fresh `properties` object on each render, so we'll need `JSON.stringify` to verify the actual contents have changed.

Alternatively we could `useMemo` on the properties but that seems like unnecessary complexity and syntax.

Reply:
Added the `JSON.stringify` to the `properties`. https://github.com/MetaMask/metamask-extension/pull/26121/commits/07e9778ddef1ca3f8e6a7a515d2016e5bb352f19